scylla/src/transport/topology.rs

```diff
@@ -743,28 +743,30 @@ impl MetadataReader {
     }
 }
 
-async fn query_metadata(
-    conn: &Arc<Connection>,
-    connect_port: u16,
-    keyspace_to_fetch: &[String],
-    fetch_schema: bool,
-) -> Result<Metadata, QueryError> {
-    let peers_query = query_peers(conn, connect_port);
-    let keyspaces_query = query_keyspaces(conn, keyspace_to_fetch, fetch_schema);
+impl ControlConnection {
+    async fn query_metadata(
+        self,
+        connect_port: u16,
+        keyspace_to_fetch: &[String],
+        fetch_schema: bool,
+    ) -> Result<Metadata, QueryError> {
+        let peers_query = self.clone().query_peers(connect_port);
+        let keyspaces_query = self.query_keyspaces(keyspace_to_fetch, fetch_schema);
```

Comment:
Commit: "topology: make metadata fetchers methods on ControlConnection "

🔧 For me it is extremely weird and unintuitive to have multiple impl blocks for a struct (ControlConnection), scattered in various places of 2 modules (control_connection and topology).
Can we please put all the `impl ControlConnection` into the `control_connection` module, and preferably make them a single `impl` block?

Reply:
https://www.reddit.com/r/rust/comments/w5l320/is_having_multiple_impl_blocks_idiomatic/

It's a perfectly valid way of logical separation. In this particular case, we separate metadata-related functionalities (defined in the impl block in `metadata.rs` main module) from the lower-level functionalities (defined in the impl block in `control_connection` module). Also, what we get is that methods defined out of `control_connection` module **can't access** mod-private fields and methods of `ControlConnection`.